i18n-report/src/main.rs

```diff
@@ -54,3 +85,89 @@ fn main() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+/// Reads each given PO file and returns message stats for each.
+fn all_stats(files: &[PathBuf]) -> anyhow::Result<Vec<MessageStats>> {
+    files
+        .iter()
+        .map(|translation| {
+            let catalog = po_file::parse(translation)
+                .with_context(|| format!("Could not parse {:?}", &translation))?;
+            let stats = MessageStats::for_catalog(&catalog);
+            Ok(stats)
+        })
+        .collect()
+}
+
+/// Prints a report showing any difference between two directories of language files.
+#[allow(clippy::print_stdout)]
+fn diff(
+    old_translations_directory: &Path,
+    new_translations_directory: &Path,
+) -> anyhow::Result<()> {
+    let mut old_translations = all_stats(&po_files(old_translations_directory)?)?;
+    let mut new_translations = all_stats(&po_files(new_translations_directory)?)?;
+    old_translations.sort_by_key(|stats| stats.language.clone());
+    new_translations.sort_by_key(|stats| stats.language.clone());
+
+    if old_translations == new_translations {
+        return Ok(());
+    }
+
+    let mut old_iter = old_translations.iter();
+    let mut new_iter = new_translations.iter();
```

Comment:
I am not very familiar with the workflows in the main repo and the way this tool is used there. Is it common that new languages are being added and removed? If that happens, I assume this report might missmatch PO files for different languages. Is that expected?

Reply:
ah, never mind, I just read the implementation and understood how it is done.